feat: add Version Support page and Commercial Support Options (#1585)
expressjs/expressjs.com: en/support/index.md

---
layout: page
title: Version Support
menu: support
lang: en
---

# Version Support

> Only the latest version of any given major release line is supported.

Reviewer: ```suggestion
Only Express 4 and Express 5 are currently supported.
```

The original sentence is true only because v5 is not released yet.  We plan to support v4 for some time after v5 is out, and later support v5 after v6 is out, etc.

It may be easier/simpler to remove this line entirely.

Author: It’s true even when 5 comes out - it’s not saying which majors are supported (that’s in the table), but eg that when v5.1 is out, v5.0 is unsupported

Reviewer: ah I see.  maybe the text can be clearer, but I'll also acquiesce if this is adjudged to be a _me_ problem

> Versions that are EOL (end of life) _may_ receive updates for critical security vulnerabilities, but the Express team offers no guarantee and has no intention of fixing any issues found or releasing any fixed versions.

Reviewer: ```suggestion
Versions that are EOL (end-of-life) _may_ receive updates for critical security vulnerabilities, but the Express team offers no guarantee and does not plan to address or release fixes for any issues found.
```

Author: Typically “EOL’ indeed means security fixes are intended to go unfixed.

Reviewer 2: Yeah I think I prefer the explicit and direct "we do not intend to patch security fixes in EOL versions".

Reviewer: I have every intention to patch _Critical_ security issues

Author: Is that the project's position?

Either way, I would think you'd want to message "no intention" regardless of your actual intention, to encourage upgrades.

Reviewer: updated suggestion as a compromise

Reviewer 2: I think it is better to under promise and over deliver, but I do agree it would be awesome if we have folks around to patch critical things as best we can.

Reviewer 2: BTW, I am good with this language. My comment was not exactly clear there. 👍 landing this suggestion but I am also comfortable with the original wording.

Author: ok, suggestion pulled in. it's semantically identical but a bit connotationally weaker, which i think is a downside, but nbd.

| Major Version | Support Start Date | Support End Date |
| -- | -- | -- |

> | [**v5.x**{: style="color: gray; opacity: 0.8" }](/{{page.lang}}/5x/api.html) | **not yet released**{: style="color: gray; opacity: 0.8" } | **not yet released**{: style="color: gray; opacity: 0.8" } |

Reviewer: `gray` color does not meet contrast requirements for a11y in dark mode

Author: what color would you suggest that's lighter than the rest?

Reviewer 3: For dark mode, I would recommend this color #abc8c3. Additionally, I think it would be better to manage the styles in the CSS file ([`styles.css`](https://github.com/expressjs/expressjs.com/blob/gh-pages/css/style.css))

Author: sure, how do i do that in jekyll?

Reviewer 3: You can add a class in the same way as a style, but in this case with `{: .my-class }`, just as I suggested in the changes.

```suggestion
| [**v5.x**{: .no-release }](/{{page.lang}}/5x/api.html) | **not yet released**{: .no-release } | **not yet released**{: .no-release } |
```

Author: done, thanks!

| [**v4.x**{: .supported }](/{{page.lang}}/4x/api.html) | April 2014 | **ongoing**{: .supported } |
| [**v3.x**{: .eol }](/{{page.lang}}/3x/api.html) | October 2012 | July 2015 |
| [**v2.x**{: .eol }](/{{page.lang}}/2x/) | March 2011 | July 2012 |
| **v1.x**{: .eol } | December 2010 | March 2011 |
| **v0.14.x**{: .eol } | December 2010 | December 2010 |

## Commercial Support Options

> If you are unable to update to the latest version of Express, please contact one of our partners to receive security updates:

Reviewer: ```suggestion
If you are unable to update to a supported version of Express, please contact one of our partners to receive security updates:
```

>  - [HeroDevs Never-Ending Support](http://www.herodevs.com/support/express-nes?utm_source=expressjs&utm_medium=link&utm_campaign=express_eol_page)

Reviewer 4: Currently this link gives a 404 but I assume is the expected until this PR is merged in our side.

Author: actually no; the link should be live now - I’ll fix that asap

update: it will be live later today.

Reviewer 4: Great!
